python/src/aiconfig/editor/server/server_utils.py

```diff
@@ -294,13 +296,12 @@ def init_server_state(
     aiconfigrc_path: str,
 ) -> Result[None, str]:
     LOGGER.info("Initializing server state for 'edit' command")
-    # TODO: saqadri - load specific .env file if specified
-    dotenv.load_dotenv()
     _load_user_parser_module_if_exists(
         initialization_settings.parsers_module_path
     )
     state = get_server_state(app)
     state.aiconfigrc_path = aiconfigrc_path
+    state.env_file_path = initialization_settings.env_file_path
```

Comment:
We should do the same validation here as on set (just move to a helper and call in both places)

Reply:
should we log to stderr if validation fails? Probably don't want to block server initialization if the env file path is invalid

Reply:
We can treat it the same way as model parsers path IMO

Reply:
Oh, I mean a helper specifically for .env validation, used both here and in set_env_file_path